SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKCodegenTool.cpp

```diff
@@ -0,0 +1,312 @@
+	CoordinatesHandler.EmitComponentReadFn = [](TypeHandler const& Handler, FString const& FieldName, int32 FieldId) {
+		return FString::Printf(TEXT("%s = Coordinates::ToFVector(GetCoordinateFromSchema(Object, %i));\n"), *FieldName, FieldId);
+	};
+	CoordinatesHandler.EmitReadFn = [](TypeHandler const& Handler, FString const& FieldName, int32 FieldId) {
+		return FString::Printf(TEXT("%s = Coordinates::ToFVector(GetCoordinateFromSchema(Object, %i));\n"), *FieldName, FieldId);
+	};
+	CoordinatesHandler.EmitWriteFn = [](TypeHandler const& Handler, FString const& FieldName, int32 FieldId) {
+		return FString::Printf(TEXT("AddCoordinateToSchema(Object, %i, Coordinates::FromFVector(%s));\n"), FieldId, *FieldName);
+	};
+
+	TypeHandler EdgeLengthHandler = CoordinatesHandler;
+	EdgeLengthHandler.SchemaTypeName = "improbable.EdgeLength";
+
+	TypesMap.Add(CoordinatesHandler.SchemaTypeName, MoveTemp(CoordinatesHandler));
+	TypesMap.Add(EdgeLengthHandler.SchemaTypeName, MoveTemp(EdgeLengthHandler));
+
+	if (CustomSchemaPath.FilePath.IsEmpty())
+	{
+		return;
+	}
+
+	FString PluginDir = FSpatialGDKServicesModule::GetSpatialGDKPluginDirectory();
+
+	FString BuildDir = FPaths::Combine(SpatialGDKServicesConstants::SpatialOSDirectory, TEXT("build"));
+	FString SchemaDir = FPaths::Combine(SpatialGDKServicesConstants::SpatialOSDirectory, TEXT("schema"));
+	FString CompiledSchemaDir = FPaths::Combine(BuildDir, TEXT("assembly/schema"));
+	FString SchemaJsonPath = FPaths::Combine(CompiledSchemaDir, TEXT("schema.json"));
+
+	IFileManager& Filesystem = IFileManager::Get();
+
+	if (!FPaths::IsUnderDirectory(CustomSchemaPath.FilePath, SchemaDir))
+	{
+		FString ErrorMessage =
+			FString::Printf(TEXT("File %s was not under directory %s. Make sure the schema file is part of the schema generation."),
+							*CustomSchemaPath.FilePath, *SchemaDir);
+		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(ErrorMessage));
+		return;
+	}
+	FString FileToConsider = CustomSchemaPath.FilePath;
+	SchemaDir.Append("/");
+	FPaths::MakePathRelativeTo(FileToConsider, *SchemaDir);
+
+	TSet<FString> Files;
+	Files.Add(FileToConsider);
+	TArray<SpatialGDK::SchemaComponent> Components;
+	SpatialGDK::ExtractComponentsDetailsFromSchemaJson(SchemaJsonPath, Components, Files);
+
+	for (const auto& Component : Components)
+	{
+		FString FieldSection;
+		FString ReadSection;
+		FString WriteSection;
+
+		bool bAllTypesHandled = true;
+
+		for (const auto& Field : Component.Fields)
+		{
+			const TypeHandler* Handler = TypesMap.Find(Field.Type);
+
+			if (Handler == nullptr)
+			{
+				FString ErrorMessage =
+					FString::Printf(TEXT("Type %s part of component %s cannot be handled."), *Field.Type, *Component.Id.Name);
+				FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(ErrorMessage));
+				bAllTypesHandled = false;
+				break;
+			}
+
+			FString FieldName = Field.Name;
+			FieldName[0] = toupper(FieldName[0]);
```

Comment:
```suggestion
			FieldName[0] = FieldName[0].ToUpper();
```

Reply:
I just want the first character to be converted, ToUpperInline seems to operate over the entire string?